integration-tests/utils/workflow-utils.js

```diff
@@ -13,27 +13,52 @@ const launchWorkflowAndWaitForSuccess = async page => {
   ])
   // stopLoggingPageAjaxResponses()
 
+  // If this table doesn't exists, something is wrong. Fail test now.
+  await page.waitForXPath('//*[@role="table" and @aria-label="submission details"]', { visible: true })
+
+  // Workflow status list:
+  // Queued -> Submitted -> Launching -> Running -> Succeeded or Failed
   const start = Date.now()
+
+  // Long enough for the submission details to be Running or Failed.
+  // Workflows might sit in "Submitted" for a long time if there is a large backlog of workflows in the environment
+  const isRunning = await Promise.race([
+    findInGrid(page, 'Failed', { timeout: 5 * 60 * 1000 }).then(() => false),
+    findInGrid(page, 'Running', { timeout: 5 * 60 * 1000 }).then(() => true)
+  ])
+
+  // If status is not Running, fails test now
+  if (!isRunning) {
+    throw new Error('Workflow has failed')
```

Comment:
If `Promise.race` call fails because it can't find `Failed` or `Running` (e.g. if workflow has not started within 5 min), error should be thrown and that should end the test.